contracts/TxRelay.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity 0.4.8;
+
+//This contract is meant as a "singleton" forwarding contract.
+//Eventually, it will be able to forward any transaction to
+//Any contract that is built to accept it.
+
+contract TxRelay {
+
+  // Note: This is a local nonce.
+  // Different from the nonce defined w/in protocol.
+  mapping(address => uint) nonce;
+
+  /*
+   * @dev Relays meta transactions
+   * @param sigV, sigR, sigS ECDSA signature on some data to be forwarded
+   * @param data The bytes necessary to call the function in the destination contract.
+                 Note, the first encoded argument in data must be address of the signer
+   * @param claimedSender Address of the user who is having tx forwarded
+   */
+  function relayMetaTx(uint8 sigV, bytes32 sigR, bytes32 sigS,
+                       address destination, bytes data,
+                       address claimedSender) {
```

Comment:
Yeah, I believe it would either take 1. a loop to copy over bytes 1-by-1 which would be very inefficient, or 2. some more assembly stuff, which it's probably good to minimize.